pydis_core/utils/paste_service.py

```diff
@@ -63,6 +623,18 @@ async def send_to_paste_service(
     if max_size > MAX_PASTE_SIZE:
         raise ValueError(f"`max_length` must not be greater than {MAX_PASTE_SIZE}")
 
+    if lexer not in SUPPORTED_LEXERS:
+        if paste_url == DEFAULT_PASTEBIN:
+            raise PasteUnsupportedLexerError(f"Lexer '{lexer}' not supported by pastebin.")
+
+        try:
+            async with http_session.get(f"{paste_url}/api/v1/lexer") as response:
+                response_json = await response.json()  # Supported lexers are the keys.
+        except Exception:
+            raise PasteUploadError("Could not fetch supported lexers from selected paste_url.")
+        if lexer not in response_json:
+            raise PasteUnsupportedLexerError(f"Lexer '{lexer}' not supported by pastebin.")
+
```

Comment:
Would it be more robust+simpler to just always send the request to get the lexers and cache that, so we don't need to keep the list of supported lexers up to date here? I doubt the extra request would be noticeable.

Reply:
Good point, added in this forcepush https://github.com/python-discord/bot-core/compare/ed2c2fad86b8083ce82d7ceee47b4b3e1e256242..240683b9a0b28ea44a05f2740af143a69cf4ac0c